### PR TITLE
Make model dependencies optional

### PR DESCRIPTION
**Pull request by znicholls**
- [x] Tests added
- [x] Documentation added
- [x] Example added (in the documentation, to an existing notebook, or in a new notebook) (N/A)
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/openscm/openscm-runner/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

File: src/openscm_runner/adapters/magicc7/magicc7.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
 import os
 from subprocess import check_output  # nosec
 
-import pymagicc
 from scmdata import ScmRun, run_append
 
 from ...progress import progress
 from ...settings import config
 from ..base import _Adapter
+from ._magicc_instances import HAS_PYMAGICC, pymagicc
```
Comment (lewisjared) [inline src/openscm_runner/adapters/magicc7/magicc7.py:13]: I like this way the best. It keeps the conditional importing in one location. Although I would refactor into a separate module. That would avoid any possible cyclic imports in future and the order of imports wouldn't be an issue (they aren't currently, but it seems cleaner)

File: setup.py [hunk 2]
```diff
@@ -70,6 +72,9 @@
     "docs": REQUIREMENTS_DOCS,
     "notebooks": REQUIREMENTS_NOTEBOOKS,
     "tests": REQUIREMENTS_TESTS,
+    "fair": REQUIREMENTS_FAIR,
```
[the remark this answers is not in the document]
Comment (znicholls) [reply setup.py:75]: You got it

File: src/openscm_runner/adapters/fair_adapter/fair_adapter.py [hunk 3]
```diff
@@ -4,16 +4,18 @@
...
 
-import fair
 import numpy as np
 import pandas as pd
 from scmdata import ScmRun
 
 from ...progress import progress
 from ..base import _Adapter
-from ._run_fair import run_fair
+from ._run_fair import HAS_FAIR, run_fair
```
[the remark this answers is not in the document]
Comment (znicholls) [reply src/openscm_runner/adapters/fair_adapter/fair_adapter.py]: Wicked thanks for the suggestion

File: src/openscm_runner/adapters/fair_adapter/fair_adapter.py [hunk 4]
```diff
@@ -4,16 +4,18 @@
...
 from ._scmdf_to_emissions import scmdf_to_emissions
 
+if HAS_FAIR:
```
[the remark this answers is not in the document]
Comment (znicholls) [reply src/openscm_runner/adapters/fair_adapter/fair_adapter.py]: As above

File: src/openscm_runner/adapters/magicc7/magicc7.py (hunk 1 above)
Comment (znicholls) [reply src/openscm_runner/adapters/magicc7/magicc7.py]: As above

File: src/openscm_runner/adapters/magicc7/_run_magicc_parallel.py [hunk 5]
```diff
@@ -7,14 +7,19 @@
...
-import f90nml
-import pymagicc
 import scmdata
 
 from ...settings import config
 from ..utils._parallel_process import _parallel_process
+from ._compat import HAS_PYMAGICC, pymagicc
 from ._magicc_instances import _MagiccInstances
 
+if HAS_PYMAGICC:
```
Comment (lewisjared) [inline src/openscm_runner/adapters/magicc7/_run_magicc_parallel.py:17]: This can come from `._compat` too

Comment (znicholls) [reply src/openscm_runner/adapters/magicc7/_run_magicc_parallel.py]: Good spot

**Comment by znicholls** [comment]
@lewisjared if you're happy hit the button

**Pull request merged**